assets/js/components/Dashboard/PermanentConfiguration/Workflows/GlEink/GlEinkWorkflow.tsx

```diff
@@ -66,24 +208,72 @@ const GlEinkWorkflow: ComponentType<WorkflowProps> = ({
         navigate(-1);
       };
       onBack = () => {
+        Object.keys(placesAndScreensToUpdate).forEach((placeId) => {
+          newScreenValidationErrors[placeId] = [];
+        });
+        Object.keys(placesAndScreensToUpdate).forEach((placeId) => {
+          pendingScreenValidationErrors[placeId] = [];
+        });
+
+        dispatch({
+          type: "SET_VALIDATION_ERRORS",
+          newScreenValidationErrors,
+          pendingScreenValidationErrors,
+        });
+        setShowValidationAlert(false);
         setConfigStep(configStep - 1);
       };
       onForward = () => {
-        putPendingScreens(
-          placesAndScreensToUpdate,
-          "gl_eink_v2",
-          configVersion
-        ).then(() => navigate("/pending"));
+        const fieldsWithErrors = validateRequiredFields(
+          placesAndScreensToUpdate
+        );
+
+        if (fieldsWithErrors.size === 0) {
+          putPendingScreens(
+            placesAndScreensToUpdate,
+            "gl_eink_v2",
+            configVersion
+          ).then((response) => {
+            handleGlEinkSubmitResponse(response, fieldsWithErrors);
+          });
+        } else {
+          validateDuplicateScreenIds(placesAndScreensToUpdate);
+          setValidationErrorMessage(generateErrorMessage(fieldsWithErrors));
+          setShowValidationAlert(true);
+          dispatch({
+            type: "SET_VALIDATION_ERRORS",
+            newScreenValidationErrors,
+            pendingScreenValidationErrors,
+          });
+        }
       };
       layout = (
-        <ConfigureScreensWorkflowPage
-          selectedPlaces={places.filter((place) =>
-            selectedPlaces.has(place.id)
+        <div>
+          <ConfigureScreensWorkflowPage
+            selectedPlaces={places.filter((place) =>
+              selectedPlaces.has(place.id)
+            )}
+            setPlacesAndScreensToUpdate={setPlacesAndScreensToUpdate}
+            handleRemoveLocation={handleRemoveLocation}
+            setConfigVersion={setConfigVersion}
+          />
+          {validationErrorMessage !== "" && (
+            <div className="config-validation-alert-container">
+              <Alert
```

Comment:
I noticed the close button hover effect looks a bit off because its positioning is towards the top instead of being centered vertically. Should the hover effect bg color be transparent to help with this, or is this by design?

Reply:
Not by design, thank you for that suggestion! Definitely looks better now